docs: "What's new in Svelte" June newsletter (#1996)
sveltejs/svelte.dev: apps/svelte.dev/content/blog/2026-06-01-whats-new-in-svelte-june-2026.md

---
title: "What's new in Svelte: June 2026"
description: 'Better forms, new long-lived remote query APIs and TypeScript 6 support in language-tools'
author: Dani Sandoval
authorURL: https://dreamindani.com
---

> This month we got a bunch of improvements in SvelteKit's forms and remote functions. Plus, a new query function (`.live(...)`) makes working with query subscriptions easier.

Reviewer: ```suggestion
This month we got a bunch of improvements in SvelteKit's forms and remote functions. Plus, a new query function (`.live(...)`) that makes accessing real-time data from the server easier.
```
Is there a reason the term "query subscription" was used here or was that the AI that came up with it? Seems like it's referred to a lot in GraphQL from a quick Google. I've suggested aligning with the language from the docs but I don't feel that strongly about it

Author: Yeah that's a weird phrase. Your suggestion is much clearer

- SvelteKit now warns when remote form validation issues are never read, helping catch missed UX paths earlier (**2.60.0**, [Docs](https://svelte.dev/docs/kit/remote-functions), [#15653](https://github.com/sveltejs/kit/pull/15653))
- Breaking: `.run()` was removed from remote queries - use `await query()` directly in all contexts instead (**2.61.0**, [Docs](https://svelte.dev/docs/kit/remote-functions#query), [#15779](https://github.com/sveltejs/kit/pull/15779))
- Remote queries can now be awaited in event handlers, async callbacks and module scope, with cache deduping shared across reactive and non-reactive consumers (**2.61.0**, [Docs](https://svelte.dev/docs/kit/remote-functions#query-Query-arguments), [#15779](https://github.com/sveltejs/kit/pull/15779))
- `query.live(...)` makes working with long-lived remote query subscriptions easier and are now async-iterable (experimental **2.59.0**, async in **2.61.0**, [Docs](https://svelte.dev/docs/kit/remote-functions#query.live), [#15878](https://github.com/sveltejs/kit/pull/15878))

> - Remote form instances now expose a programmatic `submit()` API and can be passed into `enhance` callbacks (**2.61.0**, [Docs](https://svelte.dev/docs/kit/remote-functions#form-enhance), [#15657](https://github.com/sveltejs/kit/pull/15657))

Reviewer: There was also a breaking change in the same PR worth mentioning.

> Enhance callbacks now receive a copy of the form remote function instance, rather than a `{ form, data, submit }` object.

For all the features and bugfixes that landed this month, check out the SvelteKit / Adapter [CHANGELOGs](https://github.com/sveltejs/kit/tree/main/packages).

## What's new in the Svelte ecosystem

> - Svelte language-tools now support TypeScript 6.0 across the language server, svelte2tsx and svelte-check packages (**svelte-language-server@0.18.0/svelte2tsx@0.7.55/svelte-check@4.4.8**, [Docs](https://devblogs.microsoft.com/typescript/announcing-typescript-6-0/), [#2985](https://github.com/sveltejs/language-tools/pull/2985))

Reviewer: `svelte-preprocess@6.0.4` was also updated with TS6 support if that's worth including https://github.com/sveltejs/svelte-preprocess/pull/675

> - Svelte MCP's stdio mode can now read file content directly, reducing round trips in local tool workflows (**mcp@0.1.23**, [Docs](https://svelte.dev/docs/ai/svelte-mcp), [#198](https://github.com/sveltejs/ai-tools/pull/198))

Reviewer: ```suggestion
- Svelte MCP's `stdio` mode can now read file content directly, reducing round trips in local tool workflows (**mcp@0.1.23**, [Docs](https://svelte.dev/docs/ai/svelte-mcp), [#198](https://github.com/sveltejs/ai-tools/pull/198))
```

- vite-plugin-svelte now enables the optimizer for server environments during development (**vite-plugin-svelte@7.1.0**, [#1328](https://github.com/sveltejs/vite-plugin-svelte/pull/1328))

This only covers the new features from the Svelte ecosystem. For everything else, check out the [language-tools releases](https://github.com/sveltejs/language-tools/releases), [ai-tools releases](https://github.com/sveltejs/ai-tools/releases) and [vite-plugin-svelte releases](https://github.com/sveltejs/vite-plugin-svelte/releases).

For all the bugfixes that came out in Svelte this month, you can read the full [Svelte CHANGELOG](https://github.com/sveltejs/svelte/blob/main/packages/svelte/CHANGELOG.md).

---

## Community Showcase

### Apps & Sites built with Svelte

- [Pad](https://getpad.dev) is a local-first collaboration tool that combines a Go CLI with an embedded Svelte web app for human-agent workflows ([GitHub](https://github.com/PerpetualSoftware/pad))

> - [Exort](https://github.com/Razz19/Exort/) is a local workspace for writing microcontroller code, compiling and uploading projects, and monitoring live serial output on supported boards

Reviewer: ```suggestion
- [Exort](https://www.exort.dev/) is a local workspace for writing microcontroller code, compiling and uploading projects, and monitoring live serial output on supported boards ([GitHub](https://github.com/Razz19/Exort/))
```

Author: Thank you!

- [Bingewatcher.org](https://bingewatcher.org) is a daily movie guessing game based on data from Wikipedia and word vectors for 157 languages

> - [Splitwave](https://github.com/Horuse/Splitwave) is a free node-based audio router for macOS built with Tauri + Svelte

Reviewer: ```suggestion
- [Splitwave](https://splitwave.app/) is a free node-based audio router for macOS built with Tauri + Svelte ([GitHub](https://github.com/Horuse/Splitwave))
```

- [Delcard](https://delcard.fun) is an open source peer-to-peer card game platform built with SvelteKit ([GitHub](https://github.com/LucaDeltort/delcard_games))
- [Vivix](https://vivix.dev) is a JavaScript execution visualizer that uses a worker-based interpreter for smooth timeline scrubbing ([GitHub](https://github.com/HenryOnilude/vivix))
- [hope-art.app](https://www.hope-art.app/en) applies protection filters to artwork images before sharing them online to prevent unauthorized AI training and style mimicry ([GitHub](https://github.com/HopeArtOrg/hope-re))
- [Image Palette Studio](https://image-palette-studio.vercel.app/) turns images into UI themes with generated CSS variables ([GitHub](https://github.com/aozoragh/image-palette-studio))

> - [Dialyma](https://github.com/dialymaai/dialyma) is an open source canvas builder that can export production-ready code

Reviewer: ```suggestion
- [Dialyma](https://dialyma.com/tutorials) is an open source canvas builder that can export production-ready code ([GitHub](https://github.com/dialymaai/dialyma))
```

- [Serverwat.ch](https://serverwat.ch/) is a SvelteKit dashboard for monitoring Hetzner-hosted projects

> 

Reviewer 2: I'd love to get my own in if possible : 😜 

```suggestion
- [Zenos](https://zenos.com) is a stealth startup from SvelteKit maintainer [Ben McCann](https://www.benmccann.com) that is bringing software productivity to the physical world.
```

Reviewer: I think you mean "_is_ bringing software productivity" 😆

Reviewer 2: Thanks. Gotta love auto correct. Lol

### Learning Resources

_This Week in Svelte_

> - [Ep. 140](https://www.youtube.com/watch?v=6muvGbGOG7E) - Changelog, Imperative interfaces

Reviewer: ```suggestion
- [Ep. 140](https://www.youtube.com/watch?v=AVX9lqFHOnM) - Changelog, Imperative interfaces
```

> - [Ep. 141](https://www.youtube.com/watch?v=AVX9lqFHOnM) - Changelog

Reviewer: ```suggestion
- [Ep. 141](https://www.youtube.com/watch?v=mq-52XBwUbM) - Changelog
```

> _To Read_

Reviewer 3: I'd love to get my article (https://orshot.com/blog/automate-linkedin-carousels-sveltekit) mentioned that teaches how to use SvelteKit Remote Functions to automate LinkedIn Carousels in the resources mentioned if possible - https://github.com/DreaminDani/svelte.dev/pull/5.

- [Why Svelte Is Better Than React in the Agentic Era](https://zackwebster.com/blog/why-svelte-is-better-than-react-in-the-ai-era) by Zack Webster

### Libraries, Tools & Components

> - [Huey](https://hueycolor.pages.dev) provides a composable color picker for Svelte 5

Reviewer: ```suggestion
- [Huey](https://hueycolor.pages.dev) provides a composable color picker for Svelte 5 ([GitHub](https://github.com/hueycolor/huey))
```

- [Svelte Dot Matrix Loaders](https://sv-matrix.vercel.app) provides 50+ animated dot-matrix loaders for Svelte projects ([GitHub](https://github.com/SikandarJODD/sv-matrix))
- [Paper Shaders for Svelte](https://shaders.devmischief.com) provides paper-style shader effects in an open source Svelte package ([GitHub](https://github.com/manuelogomigo/paper-shaders-svelte))

> - [EmbedPDF for Svelte](https://www.embedpdf.com/svelte-pdf-viewer) provides a headless PDF viewer for Svelte apps built on PDFium instead of PDF.js

Reviewer: ```suggestion
- [EmbedPDF for Svelte](https://www.embedpdf.com/svelte-pdf-viewer) provides a headless PDF viewer for Svelte apps built on PDFium instead of PDF.js ([GitHub](https://github.com/embedpdf/embed-pdf-viewer))
```

- [Svelte Event Calendar](https://svar.dev/demos/calendar/) is a Google-like event calendar component for Svelte with drag-and-drop editing and multiple calendar support ([GitHub](https://github.com/svar-widgets/calendar))

> - [Convex Better Auth UI for SvelteKit](https://github.com/mmailaender/Convex-Better-Auth-UI) provides self-hosted auth and organization UI components powered by Convex + Better Auth

Reviewer: ```suggestion
- [Convex Better Auth UI for SvelteKit](https://etesie.dev/docs/auth/overview/introduction) provides self-hosted auth and organization UI components powered by Convex + Better Auth ([GitHub](https://github.com/mmailaender/Convex-Better-Auth-UI))
```

> - [Aphex CMS](https://github.com/IcelandicIcecream/aphex) is an open source Sanity-inspired CMS that runs inside a single SvelteKit app

Reviewer: ```suggestion
- [Aphex CMS](https://getaphex.com/) is an open source Sanity-inspired CMS that runs inside a single SvelteKit app ([GitHub](https://github.com/IcelandicIcecream/aphex))
```

- [Svelte Hero](https://plugins.jetbrains.com/plugin/31546-svelte-hero) is a new JetBrains plugin focused on better Svelte support

> - [jscpd v4.2.0](https://github.com/kucherenko/jscpd) now supports Svelte projects for duplicate code detection ([Changelog](https://github.com/kucherenko/jscpd/blob/master/CHANGELOG.md))

Reviewer: ```suggestion
- [jscpd v4.2.0](https://jscpd.dev/) now supports Svelte projects for duplicate code detection ([Changelog](https://github.com/kucherenko/jscpd/blob/master/CHANGELOG.md#420--2026-05-14))
```
